lib/cloud/clients.go

```diff
@@ -327,6 +366,27 @@ type awsAssumeRoleOpts struct {
 	assumeRoleARN string
 	// assumeRoleExternalID is used to assume an external AWS IAM Role.
 	assumeRoleExternalID string
+
+	// credentialsSource describes which source to use to fetch credentials.
+	credentialsSource credentialsSource
+
+	// integration is the name of the integration to be used to fetch the credentials.
+	integration string
+}
+
+func (a *awsAssumeRoleOpts) checkAndSetDefaults() error {
+	switch a.credentialsSource {
+	case credentialsSourceAmbient:
+		a.integration = ""
```

Comment:
Why override the integration name instead of checking it? I'm not a 
```suggestion
		if a.integration != "" {
			return trace.BadParameter("ambient credentials shouldn't have an integration name")
		}
```

Reply:
Honestly I think we just remove the line entirely.
If it's ambient, then the integration is totally ignored anyway.

What do you think?

Reply:
I'd say that an ambient `credentialsSource` with a non empty `integration` should be an error.